docs/resources/bms_instance.md

```diff
@@ -0,0 +1,217 @@
+
+data "huaweicloud_availability_zones" "myaz" {}
+
+data "huaweicloud_vpc" "myvpc" {
+  name = "vpc-default"
+}
+
+data "huaweicloud_vpc_subnet" "mynet" {
+  name = "subnet-default"
+}
+
+data "huaweicloud_networking_secgroup" "mysecgroup" {
+  name = "default"
+}
+
+resource "huaweicloud_bms_instance" "test" {
+  name                  = var.instance_name
+  image_id              = var.image_id
+  flavor_id             = var.flavor_id
+  op_svc_userid         = var.op_svc_userid
+  security_groups       = [data.huaweicloud_networking_secgroup.mysecgroup.id]
+  availability_zone     = data.huaweicloud_availability_zones.myaz.names[0]
+  vpc_id                = data.huaweicloud_vpc.myvpc.id
+  eip_id                = huaweicloud_vpc_eip.myeip.id
+  charging_mode         = "prePaid"
+  period_unit           = "month"
+  period                = "1"
+  key_name              = var. key_name
+  enterprise_project_id = var.enterprise_project_id
+  system_disk_size      = 150
+  system_disk_type      = "SSD"
+
+  data_disks {
+    type = "SSD"
+    size = 100
+  }
+
+  nics {
+    subnet_id = data.huaweicloud_vpc_subnet.mynet.id
+    ip_address = "192.168.0.123"
+  }
+
+  tags = {
+    foo = "bar"
+    key = "value"
+  }
+}
+```
+
+`user_data` can come from a variety of sources: inline, read in from the `file`
+function, or the `template_cloudinit_config` resource.
+
+## Argument Reference
+
+-> **NOTE:** If the `user_data` field is specified for a Linux ECS that is created using an image with Cloud-Init
+installed, the `admin_pass` field becomes invalid.
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the instance.
+    If omitted, the provider-level region will be used.
+    Changing this creates a new instance.
+
+* `name` - (Required, String) Specifies a unique name for the instance.
+    The name consists of 1 to 64 characters, including letters, digits, underscores (_), hyphens (-), and periods (.).
+
+* `image_id` - (Required, String, ForceNew) Specifies the image ID of the desired image for the instance.
+    Changing this creates a new instance.
+
+* `flavor_id` - (Required, String, ForceNew) Specifies the flavor ID of the desired flavor for the instance.
+    Changing this creates a new instance.
+
+* `op_svc_userid` - (Required, String, ForceNew) Specifies the user ID. You can obtain the user ID
+    from My Credential on the management console. Changing this creates a new instance.
+
+* `user_data` - (Optional, String, ForceNew) Specifies the user data to be injected during the instance creation.
+    Text and text files can be injected.
+    Changing this creates a new server.
+
+* `admin_pass` - (Optional, String, ForceNew) Specifies the administrative password to assign to the instance.
+    Changing this creates a new instance.
+
+* `key_name` - (Optional, String, ForceNew) Specifies the name of a key pair to put on the instance.
```

Comment:
I think it is enough to name it as `keypair' and indicate in the doc that this parameter is the name of the keypair.

Reply:
fixed